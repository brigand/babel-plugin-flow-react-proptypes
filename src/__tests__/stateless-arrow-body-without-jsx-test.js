const babel = require('babel-core');
const content = `
var React = require('react');

const arrowFunctionWithBody = () => window.console;

type Choices = 'option1' | 'option2';

type FooT = {
    x?: Choices
};

const Foo = (props: FooT) => (
  React.createElement(
    'div',
    null,
    props.x
  )
);

export default Foo;
`;

it('stateless-arrow-body-without-jsx', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
