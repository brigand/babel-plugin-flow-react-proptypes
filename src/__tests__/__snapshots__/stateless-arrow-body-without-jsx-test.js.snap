// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateless-arrow-body-without-jsx 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var React = require('react');

var arrowFunctionWithBody = function arrowFunctionWithBody() {
  return window.console;
};

var Foo = function Foo(props) {
  return React.createElement('div', null, props.x);
};

Foo.propTypes = {
  x: require('prop-types').oneOf(['option1', 'option2'])
};
exports.default = Foo;"
`;
